File: questPresentation/src/main/java/ru/nekit/android/qls/quest/mediator/types/time/CurrentTimeAdapter.java
package ru.nekit.android.qls.quest.mediator.types.time;

import android.support.annotation.NonNull;
import android.view.View;

import java.util.List;

import ru.nekit.android.qls.quest.types.CurrentTimeQuest;
import ru.nekit.android.qls.utils.TimeUtils;

class CurrentTimeAdapter extends TimeAdapter {

    @NonNull
    private final CurrentTimeQuest mQuest;

    CurrentTimeAdapter(@NonNull CurrentTimeQuest quest,
                       @NonNull List<Integer> listData,
                       @NonNull View.OnClickListener clickListener) {
        super(listData, clickListener);
        mQuest = quest;
    }

    @Override
    protected int getTime(int position) {
        int delta = (int) ((TimeUtils.getCurrentTime() - mQuest.getTimeStamp()) / 1000 / 60);
        return mQuest.leftNode[position] + delta;
    }

    @Override
    public void onBindViewHolder(final TimeViewHolder holder, int position) {
        super.onBindViewHolder(holder, position);
        holder.view.setTag(position);
    }
}

File: questPresentation/src/main/java/ru/nekit/android/qls/quest/view/mediator/types/time/CurrentTimeAdapter.kt
package ru.nekit.android.qls.quest.view.mediator.types.time

import io.reactivex.subjects.Subject
import ru.nekit.android.qls.domain.model.quest.CurrentTimeQuest
import ru.nekit.android.utils.TimeUtils

//ver 1.0
class CurrentTimeAdapter(private val quest: CurrentTimeQuest, listData: List<Int>,
                         answerPublisher: Subject<Any>) : TimeAdapter(listData, answerPublisher) {

    override fun getTime(position: Int): Int {
        val delta = ((TimeUtils.currentTime - quest.timeStamp) / 1000 / 60).toInt()
        return quest.leftNode[position] + delta
    }

    override fun onBindViewHolder(holder: TimeAdapter.TimeViewHolder, position: Int) {
        super.onBindViewHolder(holder, position)
        holder.view.tag = position
    }
}